packages/roosterjs-editor-core/lib/coreApi/getContent.ts

```diff
@@ -24,15 +25,22 @@ export const getContent: GetContent = (core: EditorCore, mode: GetContentMode):
     const triggerExtractContentEvent = mode == GetContentMode.CleanHTML;
     const includeSelectionMarker = mode == GetContentMode.RawHTMLWithSelection;
 
+    // When there is fragment for shadow edit, always use the cached fragment as document since HTML node in editor
+    // has been changed by uncommited shadow edit which should be ignored.
+    const root = core.lifecycle.shadowEditFragment || core.contentDiv;
+
     if (mode == GetContentMode.PlainText) {
-        content = getTextContent(core.contentDiv);
+        content = getTextContent(root);
     } else if (triggerExtractContentEvent || core.lifecycle.isDarkMode) {
-        const clonedRoot = core.contentDiv.cloneNode(true /*deep*/) as HTMLElement;
+        const clonedRoot = cloneNode(root);
         const originalRange = core.api.getSelectionRange(core, true /*tryGetFromCache*/);
-        const path =
-            includeSelectionMarker &&
-            originalRange &&
-            getSelectionPath(core.contentDiv, originalRange);
+        const path = !includeSelectionMarker
```

Comment:
Can we wrap these into a function?

Reply:
I feel it is already clear enough to directly put it here.